### PR TITLE
Thesis linda

### PR DESCRIPTION
**Pull request by lindamaok899**
Thesis submission: shapley function and tests.

### CODE REVIEW

File: econsa/shapley.py [hunk 1]
```diff
@@ -1,9 +1,214 @@
...
+           ``random`` otherwise.
+
+    model : string
+          The model/function you will calculate the shapley effects on.
+
+    Xall : string (n)
+         A function that takes `n´ as an argument and generates a n-sample of
+         a d-dimensional input vector.
+
+    Xset : string (n, Sj, Sjc, xjc)
```
Comment (loikein) [inline econsa/shapley.py:44]: Per [PEP](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names), function or variable name should contain no upper case letter. Consider using format like `x_all`, etc.

File: econsa/shapley.py [hunk 2]
```diff
@@ -1,9 +1,214 @@
...
+    Returns
+    -------
+    effects : DataFrame
+            n dimensional DataFrame with the estimated shapley effects, the
+            standard errors and the confidence intervals for the input vectors.
+
+    Sources
+    -------
+    Shapley exact permutations: https://rdrr.io/cran/sensitivity/src/R/shapleyPermEx.R
+    Shapley random permutations: https://rdrr.io/cran/sensitivity/src/R/shapleyPermRand.R
```
Comment (loikein) [inline econsa/shapley.py:80]: Please use proper ReST links or references, check other functions, temfpy, or [reStructuredText Markup Specification](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html) if you’re not familiar with that.

**Review by loikein — CHANGES_REQUESTED** [review]
Please address the comments, and also add your functions’ docstrings to the doc’s method section, I think there was already a subsection for you. 

Also I know it’s tricky sometimes, but please try to make the tests pass. For example, you can try to run `pytest` on your local machine, and follow the warnings advices.

**Comment by lindamaok899** [comment]
> Please address the comments, and also add your functions’ docstrings to the doc’s method section, I think there was already a subsection for you.
> 
> Also I know it’s tricky sometimes, but please try to make the tests pass. For example, you can try to run `pytest` on your local machine, and follow the warnings advices.

Thanks for the review, I ran the tests using pytest on my local machine and all tests pass - I am not sure i understand the comment on the tests?

**Comment by lindamaok899** [comment]
I have integrated all the comments you have given me above. The tests are not running now because of:
1. Warnings related to the test_correlation.py module which I have not written
2. Black formatting issues which are conflicting with flake8 issues for shapley.py (it seems that for black, white space before : is okay but for flake8 it is not, when I fix the flake8 issue, I get a black error)
3. Black and flake8 issues for sampling.py (chaospy imported but not used)

Let me know if there are further problems comments.

File: econsa/shapley.py (hunk 2 above)
Comment (lindamaok899) [reply econsa/shapley.py]: done

File: econsa/shapley.py (hunk 1 above)
Comment (lindamaok899) [reply econsa/shapley.py]: done

Comment (lindamaok899) [reply econsa/shapley.py]: done

**Comment by lindamaok899** [comment]
> Please address the comments, and also add your functions’ docstrings to the doc’s method section, I think there was already a subsection for you.
> 
> Also I know it’s tricky sometimes, but please try to make the tests pass. For example, you can try to run `pytest` on your local machine, and follow the warnings advices.

done

**Comment by lindamaok899** [comment]
I have integrated all the comments you have given me above. The tests are not running now because of:

1. Warnings related to the test_correlation.py module which I have not written
2. Formatting conflict between black and flake8 for shapley.py (issue has been raised here: https://github.com/psf/black/issues/315, solution is to add a clause in conf.py that ignores the E203 error from flake8.)

Let me know if there are further problems comments.

**Comment by lindamaok899** [comment]
I have integrated all the comments you have given me above. The warnings are related to the test_correlation.py module which I have not written.

Let me know if there are further problems comments.

File: docs/source/motivation/sensitivity-analysis-quantitative.rst [hunk 3]
```diff
@@ -36,4 +36,37 @@ Again, we now apply this to the **EOQ** model. Given the current limits to our i
...
+
+.. math::
+
+  \sum_{k}^{i=1} Sh_i = Var[Y],
+
+even if there is dependence or structural interactions amongst the elements in :math:`X_K`. Throughout the package, we use :math:`Sh_i` to denote the Shapley effect and :math:`v_i` to denote the generic Shapley value.
```
Comment (loikein) [inline docs/source/motivation/sensitivity-analysis-quantitative.rst:72]: Adding a return at the end will probably solve the pre-commit error.

**Review by loikein — APPROVED** [review]

**Review by peisenha — APPROVED** [review]
Thanks, @lindamaok899 ! @loikein , please make sure to merge with master and clean up, just to be sure everyting remains properly working ... updted docs, etc.

**Pull request merged**